rasa/core/policies/flow_policy.py

```diff
@@ -721,6 +721,16 @@ def _run_step(
                         # TODO: we need to figure out how to actually "undo" the
                         #    changed slots
                         pass
+                elif not previous_flow and current_frame.flow_id != "pattern_completed":
+                    # TODO: rather than triggering the flow-trigger-action this should
+                    #   directly put the flow onto the stack and start it that way.
+                    return ActionPrediction(
+                        FLOW_PREFIX + "pattern_completed",
+                        1.0,
+                        metadata={"slots": {PREVIOUS_FLOW_SLOT: previous_flow_name}},
```

Comment:
won't this previous flow name always be None here? Ah you just resetting that slot, right? Might be more clear if you just set it to None